firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -49,42 +49,55 @@ void disableMotors()
 // Drives the motor corresponding to "motor" to it's home position (position 0)
 void homeAxis(int motor[])
 {
+  int * currentMotorPos;
+  int currentMotorDir, currentMotorNumEighthSteps;
   int i;
 
+  currentMotorPos = (motor == xMotor) ? &currentX : &currentY;
+
   // Loop until endstop collision, then fine tune it
   // LEFT and DOWN have same value, as do RIGHT and UP. Use LEFT
   // and RIGHT arbitrarily while tuning end stop for both x and y axes.
 
+  // Sets scale and direction for motor and current position
   digitalWrite(motor[DIR_PIN], LEFT);
   setScale(motor, WHOLE_STEPS);
+  currentMotorNumEighthSteps = 8;
+  currentMotorNumEighthSteps *= -1;
   while (digitalRead(motor[ENDSTOP_PIN]) == LOW)
   {
     digitalWrite(motor[STEP_PIN], LOW);
     delay(1);
     digitalWrite(motor[STEP_PIN], HIGH);
+
+    // Motor is moved above, current position for whatever motor updated here
+    *currentMotorPos += currentMotorNumEighthSteps;
   }
 
   digitalWrite(motor[DIR_PIN], RIGHT);
+  currentMotorNumEighthSteps *= -1;
```

Comment:
This moves the motor away from home to fine tune, so this should be 1, not -1

Reply:
I set this to -1 because in the first calibration the direction was set to -1, so I was multiplying it by -1 again so that it would go right. As of that point currentMotorNumEighthSteps = -8, so if I left it currentX/Y would be going the wrong direction right?

Reply:
I'm glad you challenged my comment because you're totally right, can you add a comment explaining that? It can be with the other comment for organization.